### PR TITLE
DOC: pd.Period and pd.period_range should document that they accept datetime, date and pd.Timestamp

### LINKED ISSUE #53038: DOC: `pd.Period` and `pd.period_range` should document that they accept `datetime`, `date` and `pd.Timestamp`
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.Period.html#pandas-period
https://pandas.pydata.org/docs/reference/api/pandas.period_range.html#pandas-period-range

### Documentation problem

Period documents the `value` argument as `str` or `Period`, while `period_range` will document it as `str` or `period-like`. Both just use the `Period` constructor implemented in cython internally. As per said implementation, it will also accept a `datetime`, `date` and `pd.Timestamp` as can be seen here: https://github.com/pandas-dev/pandas/blob/v2.0.1/pandas/_libs/tslibs/period.pyx#L2626-L2643

Technically it would also accept an integer/float, but this seems more of an edge case, so it might be fine to exclude it from the documentation, it's also not particularly helpful.

### Suggested fix for documentation

Extend parameter documentation with additional allowed types. It might also be nice to mention that `freq` is required if `value` is not already period-like (i.e. a Period or a string representation of a period)

This issue was opened as a result of discussion on the following pandas-stubs issue: https://github.com/pandas-dev/pandas-stubs/issues/675

### PR DESCRIPTION
**Pull request by ABizzinotto**
NOTE: This is a follow up to PR #53611 as I realised the issue was incompletely addressed there

- [x] closes #53038 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by ABizzinotto** [comment]
@mroeschke I opened this PR to follow up on #53611 in addressing #53038. I realised I didn't push one last commit to the previous PR and the additional accepted types were missing from the `end` parameter of `period_range`, I only added them to `start`. I had also added a note to `freq` on `Period` indicating it is required if `value` is `None`. 

I didn't open a new issue since this still addresses the previous one, but let me know if this is not the right course of action. Thanks and sorry!

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -2663,6 +2663,7 @@ class Period(_Period):
         One of pandas period strings or corresponding objects. Accepted
         strings are listed in the
         :ref:`offset alias section <timeseries.offset_aliases>` in the user docs.
+        If value is not already period-like, freq is required.
```
Comment (MarcoGorelli) [inline pandas/_libs/tslibs/period.pyx:2666]: are you sure this is true? e.g.
```python
In [1]: pd.Period('2020-01')
Out[1]: Period('2020-01', 'M')
```

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for your PR - just got a question

File: pandas/_libs/tslibs/period.pyx (hunk 1 above)
Comment (ABizzinotto) [reply pandas/_libs/tslibs/period.pyx]: @MarcoGorelli thanks. Good point, it works with a string, it just errors if value is None or datetime. Changed the note to reflect that.

Comment (MarcoGorelli) [reply pandas/_libs/tslibs/period.pyx]: thanks - looks fine for `None` though?
```python
In [3]: pd.Period(None)
Out[3]: NaT
```

Comment (MarcoGorelli) [reply pandas/_libs/tslibs/period.pyx]: hi - is this still active? if so, could you address this comment please?

Comment (ABizzinotto) [reply pandas/_libs/tslibs/period.pyx]: Sorry, missed this comment. I believe it should be all set now

**Review by MarcoGorelli — APPROVED** [review]
thanks @ABizzinotto

**Pull request merged**